### PR TITLE
patsh: 0.1.3 -> 0.2.0

### PR DESCRIPTION
**Pull request by figsoda**
Diff: https://github.com/nix-community/patsh/compare/v0.1.3...v0.2.0

Changelog: https://github.com/nix-community/patsh/blob/v0.2.0/CHANGELOG.md

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/tools/misc/patsh/default.nix [hunk 1]
```diff
@@ -1,30 +1,41 @@
...
-  # tests fail on darwin due to rpath issues
-  doCheck = !stdenv.isDarwin;
+  checkInputs = [ custom ];
 
-  TREE_SITTER_BASH = runCommand "tree-sitter-bash" { } ''
-    mkdir $out
-    ln -s ${tree-sitter.builtGrammars.tree-sitter-bash}/parser $out/libtree-sitter-bash.a
+  postPatch = ''
+    for file in tests/fixtures/*-expected.sh; do
+      substituteInPlace $file \
```
Comment (SuperSandro2000) [inline pkgs/development/tools/misc/patsh/default.nix:34]: If you don't know that the program is build around nix than this surprising

Comment (figsoda) [reply pkgs/development/tools/misc/patsh/default.nix:34]: should I add a comment?

Comment (SuperSandro2000) [reply pkgs/development/tools/misc/patsh/default.nix:34]: It probably wouldn't hurt

Comment (figsoda) [reply pkgs/development/tools/misc/patsh/default.nix:34]: https://github.com/NixOS/nixpkgs/pull/204676

**Pull request merged**